module/Activity/src/Activity/Service/ActivityQuery.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace Activity\Service;
+
+use Application\Service\AbstractAclService;
+use Activity\Model\Activity as ActivityModel;
+use Zend\ServiceManager\ServiceManagerAwareInterface;
+
+class ActivityQuery extends AbstractAclService implements ServiceManagerAwareInterface
+{
+    /**
+     * Get the ACL.
+     *
+     * @return \Zend\Permissions\Acl\Acl
+     */
+    public function getAcl()
+    {
+        return $this->getServiceManager()->get('activity_acl');
+    }
+
+    /**
+     * Get the default resource ID.
+     *
+     * This is used by {@link isAllowed()} when no resource is specified.
+     *
+     * @return string
+     */
+    protected function getDefaultResourceId()
+    {
+        return 'activity';
+    }
+
+
+    /**
+     * Get the information of one activity from the database.
+     *
+     * @param int $id The activity id to be searched for
+     *
+     * @return \Activity\Model\Activity Activity or null if the activity does not exist
+     */
+    public function getActivity($id)
+    {
+        if (!$this->isAllowed('view', 'activity')) {
+            $translator = $this->getTranslator();
+            throw new \User\Permissions\NotAllowedException(
+                $translator->translate('You are not allowed to view the activities')
+            );
+        }
+
+        $activityMapper = $this->getServiceManager()->get('activity_mapper_activity');
+        $activity = $activityMapper->getActivityById($id);
+
+        return $activity;
+    }
+
+    /**
+     * Get the activity with additional details
+     *
+     * @param $id
+     * @return ActivityModel
+     */
+    public function getActivityWithDetails($id)
```

Comment:
This does the same as `getActivity()`, apart from an additional permission check. Why? Is this supposed to return more details (which?) in a later iteration?
@stefanmajoor